test/test_consumer_integration.py

```diff
@@ -547,242 +538,240 @@ def test_fetch_buffer_size(self):
-        consumer = self.kafka_consumer(fetch_max_wait_ms=200,
-                                       request_timeout_ms=500)
-        tp = TopicPartition(self.topic, 0)
-        bad_tp = TopicPartition(self.topic, 100)
-
-        with self.assertRaises(ValueError):
-            consumer.offsets_for_times({tp: -1})
-
-        with self.assertRaises(KafkaTimeoutError):
-            consumer.offsets_for_times({bad_tp: 0})
+                messages.append(msg)
+    assert_message_count(messages, 5)
+    assert t.interval >= (TIMEOUT_MS / 1000.0)
+
+
+@pytest.mark.skipif(env_kafka_version() < (0, 8, 1), reason="Requires KAFKA_VERSION >= 0.8.1")
+def test_kafka_consumer__offset_commit_resume(kafka_consumer_factory, send_messages):
+    GROUP_ID = random_string(10)
+
+    send_messages(range(0, 100), partition=0)
+    send_messages(range(100, 200), partition=1)
+
+    # Start a consumer and grab the first 180 messages
+    consumer1 = kafka_consumer_factory(
+        group_id=GROUP_ID,
+        enable_auto_commit=True,
+        auto_commit_interval_ms=100,
+        auto_offset_reset='earliest',
+    )
+    output_msgs1 = []
+    for _ in range(180):
+        m = next(consumer1)
+        output_msgs1.append(m)
+    assert_message_count(output_msgs1, 180)
+
+    # Normally we let the pytest fixture `kafka_consumer_factory` handle
+    # closing as part of its teardown. Here we manually call close() to force
+    # auto-commit to occur before the second consumer starts. That way the
+    # second consumer only consumes previously unconsumed messages.
+    consumer1.close()
+
+    # Start a second consumer to grab 181-200
+    consumer2 = kafka_consumer_factory(
+        group_id=GROUP_ID,
+        enable_auto_commit=True,
+        auto_commit_interval_ms=100,
+        auto_offset_reset='earliest',
+    )
+    output_msgs2 = []
+    for _ in range(20):
+        m = next(consumer2)
+        output_msgs2.append(m)
+    assert_message_count(output_msgs2, 20)
+
+    # Verify the second consumer wasn't reconsuming messages that the first
+    # consumer already saw
+    assert_message_count(output_msgs1 + output_msgs2, 200)
+
+
+@pytest.mark.skipif(env_kafka_version() < (0, 10, 1), reason="Requires KAFKA_VERSION >= 0.10.1")
+def test_kafka_consumer_max_bytes_simple(kafka_consumer_factory, topic, send_messages):
+    send_messages(range(100, 200), partition=0)
+    send_messages(range(200, 300), partition=1)
+
+    # Start a consumer
+    consumer = kafka_consumer_factory(
+        auto_offset_reset='earliest', fetch_max_bytes=300)
+    seen_partitions = set()
+    for i in range(10):
+        poll_res = consumer.poll(timeout_ms=100)
+        for partition, msgs in poll_res.items():
+            for msg in msgs:
+                seen_partitions.add(partition)
+
+    # Check that we fetched at least 1 message from both partitions
+    assert seen_partitions == {TopicPartition(topic, 0), TopicPartition(topic, 1)}
```

Comment:
I'm seeing occasional test failures here.

I actually don't see the point of this test--when @tvoinarovskyi originally wrote it, it also tested that we were getting 3K bytes back, but that is no longer true in Kafka 0.11, so @dpkp removed that part of the test.

It appears to be verifying no partition starvation, and sometimes it works and sometimes it doesn't, but 10 poll cycles feels a little short to guarantee that we receive from both partitions... maybe increase the cycle count to 100 and not worry about the very infrequent failure?

Personally, I'd rather merge with `test_kafka_consumer_max_bytes_one_msg()` and verify that even if we are sending larger messages to two partitions that we can consume from both partitions and don't get starvation...

Perhaps this change should be saved for a separate PR and in this one just get the tests to pass by re-running the occasional failure and then merge as-is.

Reply:
Yea, I don't remember the context under which I did this test. Feels like it doesn't test much to begin with. I think I wanted to verify that large messages will pass through even with a low max_bytes.
Feel free to merge it as you see fit. Sorry for the trouble again with this test.

Reply:
Looking at your original code, the intent was good, just Kafka 0.11 changed behavior a bit so this test got munged around. I'll try to merge it with the following test in a followup PR.

Reply:
https://github.com/dpkp/kafka-python/pull/1891